server/src/main/java/org/elasticsearch/action/ingest/WriteableIngestDocument.java

```diff
@@ -107,10 +107,11 @@ IngestDocument getIngestDocument() {
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(DOC_FIELD);
-        Map<String, Object> metadataMap = ingestDocument.getMetadataMap();
-        for (Map.Entry<String, Object> metadata : metadataMap.entrySet()) {
-            if (metadata.getValue() != null) {
-                builder.field(metadata.getKey(), metadata.getValue().toString());
+        org.elasticsearch.script.Metadata metadata = ingestDocument.getMetadata();
```

Comment:
Should Metadata be able to read/write it's own XContent stream? Or maybe Metadata should have something to return a Map of its keys to values? I'm considering the case where in the future Metadata may not necessarily be backed by a Map so it won't have something like keySet and get naturally available.